data/core/titleview.lua

```diff
@@ -37,8 +37,10 @@ function TitleView:configure_hit_test(borderless)
     local icon_spacing = icon_w
     local controls_width = (icon_w + icon_spacing) * #title_commands + icon_spacing
     system.set_window_hit_test(title_height, controls_width, icon_spacing)
+    system.set_window_on_resize(core.step)
```

Comment:
I wonder if using `core.step` directly could be problematic.
When is the `event_filter` called? Could it be called while we're already in `core.step`? I worry that there could be problems with conflicting data or even just calling `renderer.begin_frame` and `renderer.end_frame` in bad moments.

Mainly this warning worries me:
```
WARNING: Be very careful of what you do in the event filter function, as it may run in a different thread!
```

Reply:
Should probably ratelimit this call as resizing window can rapid-fire this event.